05.wc/wc.rb

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options, file_paths = parse_options(ARGV)
+
+  # パイプで接続した場合 $stdin.tty? は false になる
+  if file_paths.empty? && !$stdin.tty?
+    file_stats = count_content($stdin.read)
+    file_stats[:file_path] = nil
+    file_stats_list = [file_stats]
+    total_stats = nil
+  else
+    file_stats_list, total_stats = collect_file_stats(file_paths)
+  end
+
+  max_width = calculate_max_width(file_stats_list, total_stats)
+
+  file_stats_list.each { |stats| output_format(stats, options, max_width) }
+
+  output_format(total_stats, options, max_width) if total_stats
+end
+
+def parse_options(argv)
+  opt = OptionParser.new
+  options = {}
+  opt.on('-l') { |v| options[:lines] = v }
+  opt.on('-w') { |v| options[:words] = v }
+  opt.on('-c') { |v| options[:bytes] = v }
+  file_paths = opt.parse(argv)
+  [options, file_paths]
+end
+
+def count_content(content)
+  {
+    lines: content.count("\n"),
+    words: content.split.size,
+    bytes: content.bytesize
```

Comment:
```suggestion
def count_content(content, file_path = nil)
  {
    lines: content.count("\n"),
    words: content.split.size,
    bytes: content.bytesize,
    file_path: file_path
```

この時点でfile_pathもセットしてしまうとか。（メソッド名は見直しが必要かもしれませんが）

Reply:
そこでfile_pathをセットするようにしました。それに伴い、メソッド名を `build_stats`としました。

https://github.com/cellotak/ruby-practices/pull/12/commits/551a1003f904db48c6042d18b56f71ac39ec5d9a